File: Examples/ClassDiagram/Kotlin/Library.kt
package medialibrary

/** The user's whole collection: every playlist in one place. */
class Library(val playlists: MutableList<Playlist> = mutableListOf()) {

    fun addPlaylist(playlist: Playlist) {
        playlists.add(playlist)
    }

    fun playlist(named: String): Playlist? = playlists.firstOrNull { it.name == named }
}

File: Examples/ClassDiagram/CSharp/Library.cs
using System.Collections.Generic;
using System.Linq;

namespace MediaLibrary;

public class Library {
    private readonly List<MediaItem> _items = new List<MediaItem>();
    private readonly List<Playlist> _playlists = new List<Playlist>();

    public void AddItem(MediaItem item) {
        _items.Add(item);
    }

    public void AddPlaylist(Playlist playlist) {
        _playlists.Add(playlist);
    }

    public IEnumerable<MediaItem> GetItemsByGenre(Genre genre) {
        return _items.Where(item => item.Genre == genre);
    }
    
    public IEnumerable<MediaItem> Items => _items.AsReadOnly();
    public IEnumerable<Playlist> Playlists => _playlists.AsReadOnly();
}
